### PR TITLE
75288 get all saved filters

### PR DESCRIPTION
**Pull request by llykkelarsen**
- add "defaultfilter" (bool) column to saved filters table
- alter POST endpoint to fit new defaultfilter changes
- add get saved filters endpoint

### CODE REVIEW

File: src/Equinor.Procosys.Preservation.Command/PersonCommands/CreateSavedFilter/CreateSavedFilterCommandValidator.cs [hunk 1]
```diff
@@ -12,10 +12,10 @@ public CreateSavedFilterCommandValidator(ISavedFilterValidator savedFilterValida
...
 
             RuleFor(command => command)
-                .MustAsync((command, token) => NotExistsASavedSearchWithSameTitleForPerson(command.Title, token))
+                .MustAsync((command, token) => NotExistsASavedFilterWithSameTitleForPerson(command.Title, token))
                 .WithMessage(command => $"A saved filter with this title already exists! Title={command.Title}");
 
-            async Task<bool> NotExistsASavedSearchWithSameTitleForPerson(string title, CancellationToken token)
+            async Task<bool> NotExistsASavedFilterWithSameTitleForPerson(string title, CancellationToken token)
                 => !await savedFilterValidator.ExistsWithSameTitleForPersonAsync(title, token);
         }
```
Comment (janid1967) [inline src/Equinor.Procosys.Preservation.Command/PersonCommands/CreateSavedFilter/CreateSavedFilterCommandValidator.cs:20]: I guess there should exists one and only one default filter pr. user. Validate if Default already exists for user

File: src/Equinor.Procosys.Preservation.Domain/AggregateModels/PersonAggregate/SavedFilter.cs [hunk 2]
```diff
@@ -11,14 +11,16 @@ public class SavedFilter : PlantEntityBase, ICreationAuditable
...
+        public SavedFilter(string plant, string title, string criteria, bool defaultFilter)
             : base(plant)
         {
             Title = title;
             Criteria = criteria;
+            DefaultFilter = defaultFilter;
         }
         public string Title { get; }
         public string Criteria { get; }
+        public bool DefaultFilter { get; private set; }
```
Comment (janid1967) [inline src/Equinor.Procosys.Preservation.Domain/AggregateModels/PersonAggregate/SavedFilter.cs:23]: I think property should be public and not been set via constructor. (Idea is that constructor should handle required props only)

File: src/Equinor.Procosys.Preservation.Query/GetSavedFilters/GetSavedFiltersQuery.cs [hunk 3]
```diff
@@ -0,0 +1,11 @@
+﻿using System.Collections.Generic;
+using MediatR;
+using ServiceResult;
+
+namespace Equinor.Procosys.Preservation.Query.GetSavedFilters
```
Comment (janid1967) [inline src/Equinor.Procosys.Preservation.Query/GetSavedFilters/GetSavedFiltersQuery.cs:5]: Rename ns and class to Get**All**Sa......

File: src/Equinor.Procosys.Preservation.Query/GetSavedFilters/GetSavedFiltersQueryHandler.cs [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+using System.Threading.Tasks;
+using Equinor.Procosys.Preservation.Domain;
+using Equinor.Procosys.Preservation.Domain.AggregateModels.PersonAggregate;
+using MediatR;
+using Microsoft.EntityFrameworkCore;
+using ServiceResult;
+
+namespace Equinor.Procosys.Preservation.Query.GetSavedFilters
+{
+    public class GetSavedFiltersQueryHandler : IRequestHandler<GetSavedFiltersQuery, Result<List<SavedFilterDto>>>
```
Comment (janid1967) [inline src/Equinor.Procosys.Preservation.Query/GetSavedFilters/GetSavedFiltersQueryHandler.cs:13]: GetAllSavedFiltersQueryHandler

File: src/Equinor.Procosys.Preservation.Query/GetSavedFilters/GetSavedFiltersQueryHandler.cs [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+
+            var savedFilters = person
+                .SavedFilters
+                .Select(savedFilter => new SavedFilterDto(
+                    savedFilter.Title,
+                    savedFilter.Criteria,
+                    savedFilter.DefaultFilter,
+                    savedFilter.CreatedAtUtc,
+                    savedFilter.RowVersion.ConvertToString())).ToList();
+            return new SuccessResult<List<SavedFilterDto>>(savedFilters);
```
Comment (janid1967) [inline src/Equinor.Procosys.Preservation.Query/GetSavedFilters/GetSavedFiltersQueryHandler.cs:45]: add newline before

File: src/Equinor.Procosys.Preservation.Query/GetSavedFilters/GetSavedFiltersQuery.cs [hunk 6]
```diff
@@ -0,0 +1,11 @@
...
+{
+    public class GetSavedFiltersQuery : IRequest<Result<List<SavedFilterDto>>>
+    {
+
```
Comment (janid1967) [inline src/Equinor.Procosys.Preservation.Query/GetSavedFilters/GetSavedFiltersQuery.cs:9]: remove blank

File: src/tests/Equinor.Procosys.Preservation.Command.Tests/PersonCommands/CreateSavedFilter/CreateSavedFilterCommandHandlerTests.cs [hunk 7]
```diff
@@ -53,13 +54,14 @@ public async Task HandlingCreateSavedFilterCommand_ShouldAddSavedFilterToPerson(
...
             var result = await _dut.Handle(_command, default);
-            var savedSearch = _person.SavedFilters.First();
+            var savedFilter = _person.SavedFilters.First();
 
             // Assert
             Assert.AreEqual(0, result.Errors.Count);
             Assert.AreEqual(0, result.Data);
-            Assert.AreEqual(title, savedSearch.Title);
-            Assert.AreEqual(criteria, savedSearch.Criteria);
+            Assert.AreEqual(true, savedFilter.DefaultFilter);
```
Comment (janid1967) [inline src/tests/Equinor.Procosys.Preservation.Command.Tests/PersonCommands/CreateSavedFilter/CreateSavedFilterCommandHandlerTests.cs:62]: Assert.IsTrue

File: src/tests/Equinor.Procosys.Preservation.Command.Tests/PersonCommands/CreateSavedFilter/CreateSavedFilterCommandValidatorTests.cs [hunk 8]
```diff
@@ -21,7 +21,7 @@ public void Setup_OkState()
             const string Criteria = "Criteria";
 
             _savedFilterValidatorMock = new Mock<ISavedFilterValidator>();
-            _command = new CreateSavedFilterCommand(_title, Criteria);
+            _command = new CreateSavedFilterCommand(_title, Criteria, false);
```
Comment (janid1967) [inline src/tests/Equinor.Procosys.Preservation.Command.Tests/PersonCommands/CreateSavedFilter/CreateSavedFilterCommandValidatorTests.cs:24]: undo after removing 3. param

File: src/tests/Equinor.Procosys.Preservation.Command.Tests/Validators/SavedFilterValidatorTests.cs [hunk 9]
```diff
@@ -30,7 +30,7 @@ protected override void SetupNewDatabase(DbContextOptions<PreservationContext> d
                 _personOid = new Guid();
 
                 var person = AddPerson(context, _personOid, "Current", "User");
-                var savedFilter = new SavedFilter(TestPlant, _title, Criteria);
+                var savedFilter = new SavedFilter(TestPlant, _title, Criteria, false);
```
Comment (janid1967) [inline src/tests/Equinor.Procosys.Preservation.Command.Tests/Validators/SavedFilterValidatorTests.cs:33]: undo

File: src/tests/Equinor.Procosys.Preservation.Query.Tests/GetSavedFilters/GetSavedFiltersQueryHandlerTests.cs [hunk 10]
```diff
@@ -0,0 +1,90 @@
...
+        public async Task HandleGetSavedFiltersQuery_ShouldReturnCorrectSavedFilters()
+        {
+            using (var context = new PreservationContext(_dbContextOptions, _plantProvider, _eventDispatcher,
+                _currentUserProvider))
+            {
+                var dut = new GetSavedFiltersQueryHandler(context, _currentUserProviderMock.Object);
+
+                var result = await dut.Handle(_query, default);
+                Assert.AreEqual(1, result.Data.Count);
+                Assert.IsTrue(result.Data.Any(sf => sf.Title == _title));
```
Comment (janid1967) [inline src/tests/Equinor.Procosys.Preservation.Query.Tests/GetSavedFilters/GetSavedFiltersQueryHandlerTests.cs:71]: rewrite 3 lines: 
var savedFilter = result.Data.Single
3 asserts on savedFilters instead of result.Data.Any(sf =>

File: src/tests/Equinor.Procosys.Preservation.Command.Tests/PersonCommands/CreateSavedFilter/CreateSavedFilterCommandValidatorTests.cs (hunk 8 above)
Comment (llykkelarsen) [reply src/tests/Equinor.Procosys.Preservation.Command.Tests/PersonCommands/CreateSavedFilter/CreateSavedFilterCommandValidatorTests.cs:24]: So DefaultFilter should not be via constructor in CreateSavedFilterCommand either?

Comment (janid1967) [reply src/tests/Equinor.Procosys.Preservation.Command.Tests/PersonCommands/CreateSavedFilter/CreateSavedFilterCommandValidatorTests.cs:24]: My bad. Skip this one, I resolve it

File: src/tests/Equinor.Procosys.Preservation.Query.Tests/GetAllSavedFilters/GetAllSavedFiltersQueryHandlerTests.cs [hunk 11]
```diff
@@ -46,43 +46,45 @@ protected override void SetupNewDatabase(DbContextOptions<PreservationContext> d
...
+                var dut = new GetAllSavedFiltersQueryHandler(context, _currentUserProviderMock.Object);
 
                 var result = await dut.Handle(_query, default);
+                var savedFilter = result.Data.Single();
+
                 Assert.AreEqual(1, result.Data.Count);
-                Assert.IsTrue(result.Data.Any(sf => sf.Title == _title));
-                Assert.IsTrue(result.Data.Any(sf => sf.Criteria == _criteria));
-                Assert.IsTrue(result.Data.Any(sf => sf.DefaultFilter == _defaultFilter));
+                Assert.IsTrue(savedFilter.Title == _title);
```
Comment (janid1967) [inline src/tests/Equinor.Procosys.Preservation.Query.Tests/GetAllSavedFilters/GetAllSavedFiltersQueryHandlerTests.cs:73]: AreEqual

File: src/Equinor.Procosys.Preservation.Command/PersonCommands/CreateSavedFilter/CreateSavedFilterCommandValidator.cs [hunk 12]
```diff
@@ -1,22 +1,31 @@
...
             CascadeMode = CascadeMode.StopOnFirstFailure;
 
             RuleFor(command => command)
-                .MustAsync((command, token) => NotExistsASavedSearchWithSameTitleForPerson(command.Title, token))
+                .MustAsync((command, token) => NotExistsASavedFilterWithSameTitleForPerson(command.Title, token))
                 .WithMessage(command => $"A saved filter with this title already exists! Title={command.Title}");
+            RuleFor(command => command)
+                .MustAsync((command, token) => BeAnExistingProject(command.ProjectName, token))
+                .WithMessage(command => $"Project doesn't exist! Project={command.ProjectName}");
+
```
Comment (janid1967) [inline src/Equinor.Procosys.Preservation.Command/PersonCommands/CreateSavedFilter/CreateSavedFilterCommandValidator.cs:23]: remove extra blank line if other comments

File: src/Equinor.Procosys.Preservation.Domain/AggregateModels/PersonAggregate/Person.cs [hunk 13]
```diff
@@ -48,5 +49,11 @@ public void AddSavedFilter(SavedFilter savedFilter)
 
             _savedFilters.Add(savedFilter);
         }
+
+        public SavedFilter GetDefaultFilter(string plant, int projectId)
+        {
+            var savedFil = _savedFilters.SingleOrDefault(s => s.Plant == plant && s.ProjectId == projectId && s.DefaultFilter);
```
Comment (janid1967) [inline src/Equinor.Procosys.Preservation.Domain/AggregateModels/PersonAggregate/Person.cs:55]: remove temp. variable and make method one-liner. Exp. body

File: src/Equinor.Procosys.Preservation.Domain/AggregateModels/PersonAggregate/Person.cs [hunk 14]
```diff
@@ -48,5 +49,11 @@ public void AddSavedFilter(SavedFilter savedFilter)
 
             _savedFilters.Add(savedFilter);
         }
+
+        public SavedFilter GetDefaultFilter(string plant, int projectId)
```
Comment (janid1967) [inline src/Equinor.Procosys.Preservation.Domain/AggregateModels/PersonAggregate/Person.cs:53]: Remove plant as param. SetGlobalPlantFilter should take care of plant-part of query. I guess if you test code as is, and verify output in Debug Console, you will see plant query twice. Can you check an let me know? Curious.

File: src/Equinor.Procosys.Preservation.Domain/AggregateModels/PersonAggregate/SavedFilter.cs [hunk 15]
```diff
@@ -11,14 +11,18 @@ public class SavedFilter : PlantEntityBase, ICreationAuditable
         public SavedFilter() : base(null)
         {
         }
-        public SavedFilter(string plant, string title, string criteria)
+        public SavedFilter(string plant, int projectId, string title, string criteria)
```
Comment (janid1967) [inline src/Equinor.Procosys.Preservation.Domain/AggregateModels/PersonAggregate/SavedFilter.cs:14]: Take in Project as param. Line 17 will be ProjectId = project.Id; 
Then, also add code as
            if (project.Plant != plant)
            {
                throw new ArgumentException($"Can't relate item in {project.Plant} to item in {plant}");
            }

(See Step.cs)

File: src/Equinor.Procosys.Preservation.Infrastructure/Repositories/PersonRepository.cs [hunk 16]
```diff
@@ -8,7 +8,7 @@ namespace Equinor.Procosys.Preservation.Infrastructure.Repositories
     public class PersonRepository : RepositoryBase<Person>, IPersonRepository
     {
         public PersonRepository(PreservationContext context)
-            : base(context.Persons)
+            : base(context.Persons, context.Persons.Include(p => p.SavedFilters))
```
Comment (janid1967) [inline src/Equinor.Procosys.Preservation.Infrastructure/Repositories/PersonRepository.cs:11]: revert this change so existing queries for user don't get user with savedFilters.
Add a new method ala GetWithSavedFilterByOidAsync to use in your new context.

File: src/Equinor.Procosys.Preservation.Command/PersonCommands/CreateSavedFilter/CreateSavedFilterCommandHandler.cs [hunk 17]
```diff
@@ -13,25 +14,43 @@ public class CreateSavedFilterCommandHandler : IRequestHandler<CreateSavedFilter
...
             _unitOfWork = unitOfWork;
             _plantProvider = plantProvider;
             _currentUserProvider = currentUserProvider;
+            _projectRepository = projectRepository;
         }
 
         public async Task<Result<int>> Handle(CreateSavedFilterCommand request, CancellationToken cancellationToken)
         {
             var currentUserOid = _currentUserProvider.GetCurrentUserOid();
             var person = await _personRepository.GetByOidAsync(currentUserOid);
```
Comment (janid1967) [inline src/Equinor.Procosys.Preservation.Command/PersonCommands/CreateSavedFilter/CreateSavedFilterCommandHandler.cs:36]: use new method GetWithSavedFilterByOidAsync as mention in other comment

File: src/Equinor.Procosys.Preservation.WebApi/Controllers/Persons/CreateSavedFilterDtoValidator.cs [hunk 18]
```diff
@@ -7,6 +8,10 @@ public class CreateSavedFilterDtoValidator : AbstractValidator<CreateSavedFilter
     {
         public CreateSavedFilterDtoValidator()
         {
+            RuleFor(x => x.ProjectName)
+                .NotNull()
+                .NotEmpty()
+                .MaximumLength(Project.NameLengthMax);
```
Comment (janid1967) [inline src/Equinor.Procosys.Preservation.WebApi/Controllers/Persons/CreateSavedFilterDtoValidator.cs:14]: not necessary to check MaximumLength here since we don't add new projects in this context

File: src/Equinor.Procosys.Preservation.WebApi/Controllers/Persons/PersonsController.cs [hunk 19]
```diff
@@ -26,7 +27,18 @@ public class PersonsController : ControllerBase
...
             [FromBody] CreateSavedFilterDto dto)
         {
-            var result = await _mediator.Send(new CreateSavedFilterCommand(dto.Title, dto.Criteria));
+            var result = await _mediator.Send(new CreateSavedFilterCommand(dto.ProjectName, dto.Title, dto.Criteria, dto.DefaultFilter));
+            return this.FromResult(result);
+        }
+         
+        [Authorize(Roles = Permissions.PRESERVATION_READ)]
+        [HttpGet("/SavedFilters")]
+        public async Task<ActionResult<List<SavedFilterDto>>> GetAllSavedFilters(
```
Comment (janid1967) [inline src/Equinor.Procosys.Preservation.WebApi/Controllers/Persons/PersonsController.cs:36]: Should this be GetSavedFiltersInProject with projectName as param?

File: src/Equinor.Procosys.Preservation.Command/PersonCommands/CreateSavedFilter/CreateSavedFilterCommandValidator.cs [hunk 20]
```diff
@@ -1,22 +1,31 @@
...
+                .MustAsync((command, token) => NotExistsASavedFilterWithSameTitleForPerson(command.Title, token))
                 .WithMessage(command => $"A saved filter with this title already exists! Title={command.Title}");
+            RuleFor(command => command)
+                .MustAsync((command, token) => BeAnExistingProject(command.ProjectName, token))
+                .WithMessage(command => $"Project doesn't exist! Project={command.ProjectName}");
+             
 
-            async Task<bool> NotExistsASavedSearchWithSameTitleForPerson(string title, CancellationToken token)
+            async Task<bool> NotExistsASavedFilterWithSameTitleForPerson(string title, CancellationToken token)
                 => !await savedFilterValidator.ExistsWithSameTitleForPersonAsync(title, token);
```
Comment (janid1967) [inline src/Equinor.Procosys.Preservation.Command/PersonCommands/CreateSavedFilter/CreateSavedFilterCommandValidator.cs:26]: ExistsWithSameTitleForPersonInProjectAsync and add project to query

File: src/Equinor.Procosys.Preservation.Command/Validators/SavedFilterValidators/SavedFilterValidator.cs [hunk 21]
```diff
@@ -12,26 +13,32 @@ public class SavedFilterValidator : ISavedFilterValidator
         private readonly IReadOnlyContext _context;
         private readonly IPersonRepository _personRepository;
```
Comment (janid1967) [inline src/Equinor.Procosys.Preservation.Command/Validators/SavedFilterValidators/SavedFilterValidator.cs:14]: Do not use repos in Validators. Use context. See other Validators, none has repos
This because repos has change-tracking and are to be used in command handlers when we end up with as save.

File: src/Equinor.Procosys.Preservation.Command/Validators/SavedFilterValidators/SavedFilterValidator.cs [hunk 22]
```diff
@@ -12,26 +13,32 @@ public class SavedFilterValidator : ISavedFilterValidator
...
             var currentUser = await _personRepository.GetByOidAsync(_currentUserProvider.GetCurrentUserOid());
 
             var person = await (from p in _context.QuerySet<Person>()
                     .Include(p => p.SavedFilters)
                 where p.Id == currentUser.Id
-                select p).SingleOrDefaultAsync(token); 
-            return person.SavedFilters.Any(p => p.Title == title);
+                select p).SingleOrDefaultAsync(token);
+
+            var project = _projectRepository.GetProjectOnlyByNameAsync(projectName);
```
Comment (janid1967) [inline src/Equinor.Procosys.Preservation.Command/Validators/SavedFilterValidators/SavedFilterValidator.cs:39]: get project earlier (using context). Handle if project not found (which will cause with a nullpointer exception in your code). Getting project early, and you can use project.id in person query at line 34

File: src/Equinor.Procosys.Preservation.Query/GetAllSavedFilters/GetAllSavedFiltersQuery.cs [hunk 23]
```diff
@@ -0,0 +1,13 @@
+﻿using System.Collections.Generic;
+using Equinor.Procosys.Preservation.Domain;
+using MediatR;
+using ServiceResult;
+
+namespace Equinor.Procosys.Preservation.Query.GetAllSavedFilters
+{
+    public class GetAllSavedFiltersQuery : IRequest<Result<List<SavedFilterDto>>>, IProjectRequest
```
Comment (janid1967) [inline src/Equinor.Procosys.Preservation.Query/GetAllSavedFilters/GetAllSavedFiltersQuery.cs:8]: rename? GetSavedFiltersInProjectQuery

File: src/Equinor.Procosys.Preservation.Query/GetAllSavedFilters/GetAllSavedFiltersQueryHandler.cs [hunk 24]
```diff
@@ -0,0 +1,56 @@
...
+    public class GetAllSavedFiltersQueryHandler : IRequestHandler<GetAllSavedFiltersQuery, Result<List<SavedFilterDto>>>
+    {
+        private readonly IReadOnlyContext _context;
+        private readonly ICurrentUserProvider _currentUserProvider;
+        private readonly IProjectRepository _projectRepository;
+
+        public GetAllSavedFiltersQueryHandler(
+            IReadOnlyContext context,
+            ICurrentUserProvider currentUserProvider,
+            IProjectRepository projectRepository)
```
Comment (janid1967) [inline src/Equinor.Procosys.Preservation.Query/GetAllSavedFilters/GetAllSavedFiltersQueryHandler.cs:23]: Do not use repos in Queries (same reason as I said in Validators). Use context.

File: src/Equinor.Procosys.Preservation.Query/GetAllSavedFilters/GetAllSavedFiltersQueryHandler.cs [hunk 25]
```diff
@@ -0,0 +1,56 @@
...
+            _currentUserProvider = currentUserProvider;
+            _projectRepository = projectRepository;
+        }
+
+        public async Task<Result<List<SavedFilterDto>>> Handle(GetAllSavedFiltersQuery request,
+            CancellationToken cancellationToken)
+        {
+            var currentUserOid = _currentUserProvider.GetCurrentUserOid();
+
+            var person = await
```
Comment (janid1967) [inline src/Equinor.Procosys.Preservation.Query/GetAllSavedFilters/GetAllSavedFiltersQueryHandler.cs:35]: join in SavedFilters + Project and add where-clause handling project == request.ProjectName

File: src/Equinor.Procosys.Preservation.Query/GetAllSavedFilters/GetAllSavedFiltersQueryHandler.cs [hunk 26]
```diff
@@ -0,0 +1,56 @@
...
+                (from p in _context.QuerySet<Person>()
+                        .Include(p => p.SavedFilters)
+                    where p.Oid == currentUserOid
+                    select p).SingleOrDefaultAsync(cancellationToken);
+
+            var project = await _projectRepository.GetProjectOnlyByNameAsync(request.ProjectName);
+
+            var savedFilters = person
+                .SavedFilters
+                .Where(sf => sf.ProjectId == project.Id)
```
Comment (janid1967) [inline src/Equinor.Procosys.Preservation.Query/GetAllSavedFilters/GetAllSavedFiltersQueryHandler.cs:45]: remove .Where here

File: src/Equinor.Procosys.Preservation.Domain/AggregateModels/PersonAggregate/Person.cs [hunk 27]
```diff
@@ -48,5 +49,8 @@ public void AddSavedFilter(SavedFilter savedFilter)
```
Comment (janid1967) [inline src/Equinor.Procosys.Preservation.Domain/AggregateModels/PersonAggregate/Person.cs:49]: Check if filter and person has same plant. See AddStep

File: src/Equinor.Procosys.Preservation.Query/GetAllSavedFilters/GetAllSavedFiltersQueryHandler.cs [hunk 28]
```diff
@@ -24,33 +23,29 @@ public class GetAllSavedFiltersQueryHandler : IRequestHandler<GetAllSavedFilters
...
-                (from p in _context.QuerySet<Person>()
-                        .Include(p => p.SavedFilters)
-                    where p.Oid == currentUserOid
-                    select p).SingleOrDefaultAsync(cancellationToken);
+            var savedFilters = await (from s in _context.QuerySet<SavedFilter>()
+                join p in _context.QuerySet<Person>() on EF.Property<int>(s, "PersonId") equals p.Id
+                join pr in _context.QuerySet<Project>() on s.ProjectId equals pr.Id
+                where pr.Name == request.ProjectName
+                      && p.Oid == currentUserOid
+                select p.SavedFilters).SingleOrDefaultAsync(cancellationToken);
```
Comment (janid1967) [inline src/Equinor.Procosys.Preservation.Query/GetAllSavedFilters/GetAllSavedFiltersQueryHandler.cs:38]: Shouldn't this be select s).ToListAsync?

File: src/Equinor.Procosys.Preservation.Query/GetAllSavedFilters/GetAllSavedFiltersQueryHandler.cs [hunk 29]
```diff
@@ -0,0 +1,51 @@
...
+{
+    public class GetAllSavedFiltersQueryHandler : IRequestHandler<GetAllSavedFiltersQuery, Result<List<SavedFilterDto>>>
+    {
+        private readonly IReadOnlyContext _context;
+        private readonly ICurrentUserProvider _currentUserProvider;
+
+        public GetAllSavedFiltersQueryHandler(
+            IReadOnlyContext context,
+            ICurrentUserProvider currentUserProvider,
+            IProjectRepository projectRepository)
```
Comment (janid1967) [inline src/Equinor.Procosys.Preservation.Query/GetAllSavedFilters/GetAllSavedFiltersQueryHandler.cs:22]: remove

File: src/tests/Equinor.Procosys.Preservation.Query.Tests/GetAllSavedFilters/GetAllSavedFiltersQueryHandlerTests.cs [hunk 30]
```diff
@@ -0,0 +1,104 @@
...
+using ServiceResult;
+
+namespace Equinor.Procosys.Preservation.Query.Tests.GetAllSavedFilters
+{
+    [TestClass]
+    public class GetAllSavedFiltersQueryHandlerTests : ReadOnlyTestsBase
+    {
+        private GetAllSavedFiltersQuery _query;
+        private Mock<ICurrentUserProvider> _currentUserProviderMock;
+        private Mock<IProjectRepository> _projectRepositoryMock;
```
Comment (janid1967) [inline src/tests/Equinor.Procosys.Preservation.Query.Tests/GetAllSavedFilters/GetAllSavedFiltersQueryHandlerTests.cs:22]: remove

File: src/Equinor.Procosys.Preservation.Domain/AggregateModels/PersonAggregate/Person.cs (hunk 27 above)
Comment (llykkelarsen) [reply src/Equinor.Procosys.Preservation.Domain/AggregateModels/PersonAggregate/Person.cs:49]: Person doesn't have PlantEntityBase, just entitybase. Do I change this?
